src/components/cardList/CardList.jsx

```diff
@@ -0,0 +1,175 @@
+
+    if (this.props.fetchListActionCreator !== undefined) {
+      this.props.fetchListActionCreator();
+
+      if (this.props.delayInterval !== undefined && this.props.delayInterval > 0) {
+        this.interval = setInterval(() =>
+          this.props.fetchListActionCreator(), this.props.delayInterval);
+      }
+    }
+
+    if (this.props.onMountOtherActionCreators !== undefined) {
+      this.props.onMountOtherActionCreators.forEach((action) =>action());
+    }
+
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.interval);
+  }
+
+  applyFiltersToList(list) {
+    let filters = this.props.filters ? [...this.props.filters] : [];
+
+    // add any optional filters
+    if (this.state.filters) {
+      filters =
+        [...filters,
+          ...(this.state.filters
+            .filter((v) => v.enabled)
+            .map((v) => v.filter))
+        ];
+    }
+
+    return applyFilters(list, filters, this.props.optionalFiltersType);
+  }
+
+  handleFilterToggle(e) {
+    this.setState((state) => {
+      state.filters = state.filters
+        .map((filter) => {
+          if ( e.includes(filter.key) ) {
+            filter.enabled = true;
+          } else {
+            filter.enabled = false;
+          }
+          return filter;
+        });
+
+      return state;
+    });
+  }
+
+  onRowSelected(row) {
+    if (this.props.rowSelectedActionCreators) {
+      this.props.rowSelectedActionCreators.forEach((f) => this.props.dispatch(f(row)));
+    }
+  }
+
+  render() {
+
+    const filterCheckboxes = this.state.filters.map((filter, index) => {
+      return (
+        <ToggleButton value={ filter.key } key={index}>
+          {filter.label}
+        </ToggleButton>
+      );
+    });
+
+    const filterButtons = (
+      <div>
+        <ButtonToolbar>
+          <ToggleButtonGroup type="checkbox" onChange={ (e) => this.handleFilterToggle(e) }>
+            { filterCheckboxes }
+          </ToggleButtonGroup>
+        </ButtonToolbar>
+      </div>
+      );
+
+    return (
+      <div>
+        <h3><Label>{this.props.title}</Label></h3>
+        <h3><Label>{''}</Label></h3>
+        {/* <DataGrid
+          columnDefs={this.props.columnDefs}
+          loading={this.props.loading}
+          rowData={this.applyFiltersToList(this.props.rowData)}
+          onRowCount={this.props.onRowCount}
+          rowSelectedActionCreators={this.props.rowSelectedActionCreators}
+          filters = { this.props.optionalFilters ? filterButtons : undefined }
+        /> */}
+        {this.props.filters &&
+          <div className="filters">
+            <span>Filter:</span>
+            <span>{ this.props.optionalFilters ? filterButtons : undefined }</span>
+          </div>
+        }
+        {this.applyFiltersToList(this.props.rowData).map((patient, index) => (
+          <div 
+            className="card-list"
+            key={index} 
+            onClick={() => this.onRowSelected(patient)}>
+            <div className="left-items">
```

Comment:
@jomadoye I went ahead and merged this , but as a follow up can we extract with div block into a new PatientCard element you can pass into the CardList, so that other users can define and use their own cards?